src/v0/destinations/amazon_audience/utils.js

```diff
@@ -115,75 +117,67 @@ const batchEvents = (responseList, destination) => {
-  // using undefined as fallback so in case properties are not string
-  if (email) {
-    const em =
-      email
-        ?.replace(/[^\d.@A-Za-z-]/g, '')
-        ?.trim()
-        ?.toLowerCase() || undefined;
-    user.email = enableHash ? sha256(em) : em;
+  if (country) {
+    const country_code = formatter.formatCountry(country);
+    user.country = enableHash ? sha256(country_code) : country;
+    if (phone_number) {
+      user.phone = enableHash
+        ? sha256(formatter.formatPhone(phone_number, country_code))
+        : phone_number;
+    }
   }
-  if (phone) {
-    const phoneNumber = phone.replace(/\D/g, '').replace(/^0+/, '');
-    user.phone = enableHash ? sha256(phoneNumber) : phoneNumber;
+  // in case phone_number is present in hashed format and no country is available
+  if (phone_number && !enableHash) {
+    user.phone = phone_number;
   }
+  if (email) {
+    user.email = enableHash ? sha256(formatter.formatEmail(email)) : email;
+  }
+
   if (state) {
-    const st =
-      state
-        ?.replace(/[^\dA-Za-z]/g, '')
-        ?.trim()
-        ?.toLowerCase() || undefined;
-    user.state = enableHash ? sha256(st) : st;
+    user.state = enableHash ? sha256(formatter.formatState(state, country)) : state;
   }
   if (city) {
-    const ct =
-      city
-        ?.replace(/[^\dA-Za-z]/g, '')
-        ?.trim()
-        ?.toLowerCase() || undefined;
-    user.city = enableHash ? sha256(ct) : ct;
+    user.city = enableHash ? sha256(formatter.formatCity(city)) : city;
   }
   if (firstName) {
-    const fn =
-      firstName
-        ?.replace(/[^\dA-Za-z]/g, '')
-        ?.trim()
-        ?.toLowerCase() || undefined;
-    user.firstName = enableHash ? sha256(fn) : fn;
+    user.firstName = enableHash ? sha256(formatter.formatName(firstName)) : firstName;
   }
   if (lastName) {
-    const ln =
-      lastName
-        ?.replace(/[^\dA-Za-z]/g, '')
-        ?.trim()
-        ?.toLowerCase() || undefined;
-    user.lastName = enableHash ? sha256(ln) : ln;
+    user.lastName = enableHash ? sha256(formatter.formatName(lastName)) : lastName;
   }
   if (address) {
-    // removing extra whitespaces, non alphanumeric char, accents and doing lowercasing of the result string
-    const add =
+    const formatted_address =
       address
         ?.normalize('NFD')
         ?.replace(/[\u0300-\u036f]/g, '')
         ?.trim()
         ?.toLowerCase()
         .replace(/[^\dA-Za-z]/g, '') || undefined;
-    user.address = enableHash ? sha256(add) : add;
+    user.address = enableHash
+      ? sha256(formatter.formatAddress(formatted_address, country))
+      : address;
   }
   if (postalCode) {
-    const zip =
-      postalCode
-        ?.replace(/[^\dA-Za-z]/g, '')
-        ?.trim()
-        ?.toLowerCase()
-        ?.substring(0, postalCode.length - 4) || undefined;
-    user.postalCode = enableHash ? sha256(zip) : zip;
+    user.postalCode = enableHash ? sha256(formatter.formatPostal(postalCode)) : postalCode;
```

Comment:
Rather than checking is hashing enabled for every scenario check if hashing is enabled at first step

if yes map them as is
Else do all the formatting